### ISSUE TITLE
Mask action for policy gradient

### ISSUE THREAD

**Issue by 127161782** [aai-institute/tianshou#542]
DQN has mask mechanism:https://github.com/thu-ml/tianshou/blob/master/tianshou/policy/modelfree/dqn.py in forward Function. But. the pg strategy does not seem to have a related mechanism for mask processing. If so, can it be added? Below is my application code:

```python
# collector
def preprocess_fn(**kwargs):
    batch = Batch()
    batch.update(kwargs)
    # return batch
    if "rew" not in kwargs:
        return batch
    else:
        batch.obs = {"obs": batch["obs_next"], "mask": kwargs["info"]["mask"]}
        return batch

# train_collector = Collector(policy, train_envs, buf, preprocess_fn=preprocess_fn, exploration_noise=True)
```

**Comment by Trinkle23897** [comment]
It's better to modify the forward function in PGPolicy. I'm not sure how exactly to change the code...I'll figure it out later.

**Comment by 127161782** [comment]
OK，I am waiting

**Comment by Trinkle23897** [comment]
In `policy/modelfree/pg.py`:

```diff
        logits, hidden = self.actor(batch.obs, state=state)
        if isinstance(logits, tuple):
+           # this is for (mu, sigma) from Normal distribution
            dist = self.dist_fn(*logits)
        else:  # categorical distribution
+           # mask: np.array with shape=(bsz, n_act), dtype=bool, True means available
+           # logits: torch.Tensor with shape=(bsz, n_act), dtype=torch.float, range=0~1
+           logits *= to_torch_as(mask, logits)
            dist = self.dist_fn(logits)
```
